src/main/java/org/dependencytrack/util/NotificationUtil.java

```diff
@@ -272,6 +273,25 @@ public static JsonObject toJson(final Project project) {
         return projectBuilder.build();
     }
 
+    public static JsonObject toJson(final UserPrincipal user) {
+        final JsonObjectBuilder userBuilder = Json.createObjectBuilder();
+
+        userBuilder.add("username", user.getUsername());
+
+        if (user.getId() != 0) {
+            userBuilder.add("id", user.getId());
+        }
```

Comment:
The ID is purely internal and should not be part of the notification IMO.

Reply:
the problem is that after the user is deleted all the other fields are null